tests/test_Halem.py

```diff
@@ -507,3 +508,295 @@ def compute_v_provider(v_empty, v_full):
+    }  # The coordinates of the project site
+    data_to_site = {
+        "env": my_env,  # The simpy environment defined in the first cel
+        "name": "Dumplocatie",  # The name of the site
+        "geometry": [],  # The coordinates of the project site
+        "capacity": 5_000,  # The capacity of the site
+        "level": 0,
+    }  # The actual volume of the site (empty of course)
+    path = [start, stop]
+    Nodes, Edges = connect_sites_with_path(
+        data_from_site, data_to_site, data_node, path
+    )
+    FG = nx.Graph()
+    positions = {}
+    for node in Nodes:
+        positions[node.name] = (node.geometry.x, node.geometry.y)
+        FG.add_node(node.name, geometry=node.geometry)
+    for edge in Edges:
+        FG.add_edge(edge[0].name, edge[1].name, weight=1)
+    TransportProcessingResource = type(
+        "TransportProcessingResource",
+        (
+            core.Identifiable,  # Give it a name
+            core.Log,  # Allow logging of all discrete events
+            core.ContainerDependentMovable,  # A moving container, so capacity and location
+            core.Processor,  # Allow for loading and unloading
+            core.LoadingFunction,
+            core.UnloadingFunction,
+            core.HasResource,  # Add information on serving equipment
+            core.Routeable,
+        ),  # Initialize spill terms
+        {},
+    )
+
+    def compute_v_provider(v_empty, v_full):
+        return lambda x: x * (v_full - v_empty) + v_empty
+
+    route = []
+
+    data_hopper = {
+        "env": my_env,  # The simpy environment
+        "name": "Hopper 01",  # Name
+        "geometry": Nodes[0].geometry,  # It starts at the "from site"
+        "loading_rate": 1.5,  # Loading rate
+        "unloading_rate": 1.5,  # Unloading rate
+        "capacity": 5_000,  # Capacity of the hopper - "Beunvolume"
+        "compute_v": compute_v_provider(7, 5),  # Variable speed
+        "route": route,
+        "optimize_route": True,  # Optimize the Route
+        "optimization_type": "time",  # Optimize for the fastest path
+    }
+    hopper = TransportProcessingResource(**data_hopper)
+    activity = model.Activity(
+        env=my_env,  # The simpy environment defined in the first cel
+        name="Soil movement",  # We are moving soil
+        origin=Nodes[0],  # We originate from the from_site
+        destination=Nodes[-1],  # And therefore travel to the to_site
+        loader=hopper,  # The benefit of a TSHD, all steps can be done
+        mover=hopper,  # The benefit of a TSHD, all steps can be done
+        unloader=hopper,  # The benefit of a TSHD, all steps can be done
+        start_event=None,  # We can start right away
+        stop_event=None,
+    )  # We stop once there is nothing more to move
+    my_env.FG = FG
+    my_env.Roadmap = Roadmap
+    my_env.run()
+
+    opt_LF = len(
+        list(set(np.array(hopper.log["Value"])[np.array(hopper.log["Value"]) > 0]))
+    )
+
+    assert opt_LF < no_opt_LF
```

Comment:
Add some comments to explain what your test is about

Reply:
Done for all the HALEM tests in commit "implement change requests for pull request"